modules/apps/frontend-data-set/frontend-data-set-web/src/main/resources/META-INF/resources/views/cards/Cards.tsx

```diff
@@ -81,37 +61,61 @@ const Card = ({item, schema}) => {
 						});
 					}
 				},
-				symbolLeft: action.icon,
 			}))}
 			description={localizedDescription}
 			href={(schema.link && item[schema.link]) || null}
-			imgProps={schema.image && item[schema.image]}
+			imgProps={imageProps}
 			onSelectChange={
-				selectable && (() => selectItems(item[selectedItemsKey]))
+				selectable
+					? () => {
+							selectItems(selectedItemKey);
+					  }
+					: undefined
 			}
-			selected={
-				selectable &&
-				!!selectedItemsValue.find(
-					(element) => element === item[selectedItemsKey]
-				)
-			}
-			stickerProps={schema.sticker && item[schema.sticker]}
+			selected={cardSelected}
+			stickerProps={null}
 			symbol={schema.symbol && item[schema.symbol]}
 			title={localizedTitle}
 		/>
 	);
 };
 
-Cards.propTypes = {
-	items: PropTypes.array,
-	schema: PropTypes.shape({
-		description: PropTypes.string,
-		imgProps: PropTypes.imgProps,
-		labels: PropTypes.arrayOf(PropTypes.string),
-		link: PropTypes.string,
-		stickerProps: PropTypes.string,
-		title: PropTypes.string,
-	}).isRequired,
+const Cards = ({items, schema}: {items: Array<any>; schema: ICardSchema}) => {
+	const {selectedItemsKey, style}: IFrontendDataSetContext = useContext(
+		FrontendDataSetContext
+	);
+
+	return items?.length ? (
+		<div
+			className={classNames(
+				'cards-container mb-n4',
+				style === 'default' && 'px-3 pt-4'
+			)}
+		>
+			<div className="row">
+				{items.map((item) => {
+					return (
+						<div
+							className="col-md-3"
+							key={
+								selectedItemsKey
+									? item[selectedItemsKey]
+									: getRandomId()
+							}
+						>
+							<Card item={item} schema={schema} />
+						</div>
+					);
+				})}
+			</div>
+		</div>
+	) : (
+		<ClayEmptyState
+			description={Liferay.Language.get('sorry,-no-results-were-found')}
+			imgSrc={`${Liferay.ThemeDisplay.getPathThemeImages()}/states/search_state.gif`}
+			title={Liferay.Language.get('no-results-found')}
+		/>
```

Comment:
@dsanz We should look into removing this empty state check, along with [equivalents in other views](https://github.com/liferay/liferay-portal/blob/master/modules/apps/frontend-data-set/frontend-data-set-web/src/main/resources/META-INF/resources/views/table/Table.js#L204-L212). We are already checking `items?.length`, along with other props, in the [parent component](https://github.com/liferay/liferay-portal/blob/master/modules/apps/frontend-data-set/frontend-data-set-web/src/main/resources/META-INF/resources/FrontendDataSet.js#L568-L598). This check only makes sense if the two props in parent component make sense:
-  `overrideEmptyResultView`. Not used anywhere in DXP. Why would anybody choose not to have a nice empty state?
- `inlineAddingSettings`. Not used anywhere in DXP. This is a part of never-finished and very broken inline editing. 

IMO, they don't make sense. We should remove those two props, and simplify implementations. If we do want to keep them, we should update empty state resolution in Views to use `emptyState` prop instead of hardcodes.

Not a blocker for this PR, this would be a separate task.

Reply:
yes I've realized about this when working with Cards. I'll check it when adding https://github.com/liferay-frontend/liferay-portal/pull/4109 on top of this